src/Login.ts

```diff
@@ -91,8 +90,12 @@ export default class Login {
 
     public async getFlows(): Promise<Array<LoginFlow>> {
         const client = this.createTemporaryClient();
-        const { flows } = await client.loginFlows();
-        this.flows = flows;
+        const { flows }: { flows: LoginFlow[] } = await client.loginFlows();
+        // If an m.login.sso flow is present that is flagged as being for MSC3824 OIDC compatibility then we only return that flow
```

Comment:
Sorry, I can't remember the reasoning for this?

Reply:
I've added some extra details in the inline comment to address this.